app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/TrackListViewModel.kt

```diff
@@ -29,10 +31,12 @@ constructor(
   }
 
   private fun observeTracks() {
-    CoroutineScope(Dispatchers.IO).launch {
-      repository.getAll().collect { tracks ->
-        _uiState.value = UiState(tracks = tracks, loading = false)
-      }
+    viewModelScope.launch {
+      trackRepository.getAll().collect {
+        _uiState.value = _uiState.value.copy(tracks = it, loading = false)
+      } // TODO : update to get only useful tracks
```

Comment:
So this is for the TrackList that just displays ALL tracks right? Then, I think the TODO is not necessary... although the screen itself is kind of obsolete once we have proper beacon- and user-specific track lists.

Reply:
I mean, it was more todo : make a function that can fetch tracks that are only from a certain beacon etc.... It's true that it will be obsolete soon